ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/VersusBuyAndHoldCriterionTest.java

```diff
@@ -37,7 +37,14 @@
 public class VersusBuyAndHoldCriterionTest extends AbstractCriterionTest{
 
     public VersusBuyAndHoldCriterionTest(Function<Number, Num> numFunction) {
-        super(numFunction);
+        super(
+            (params) ->
+                params.length == 0
+                    ? new VersusBuyAndHoldCriterion(new TotalProfitCriterion())
+                    : new VersusBuyAndHoldCriterion((AnalysisCriterion)params[0])
+            ,
+            numFunction
+        );
```

Comment:
This doesn't look right to me.  I'd rather remove the conditional:
```
super(
    (params) -> new VersusBuyAndHoldCriterion((AnalysisCriterion) params[0]),
    numFunction);
 ```
And change the `getCriterion()` calls to `getCriterion(new TotalProfitCriterion())`.

Reply:
Done 👍